### PR TITLE
Conveyor code improvement

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Code improvement to some funny conveyor code to be less guh.
Removes some unused and/or broken code, and generally cleans up a bit.

Originally I was going to change how conveyors worked but then decided that the overall effect would look worse so, just a refactor.
## Why It's Good For The Game
Less shitty code.
## Changelog
:cl:
code: Cleaned up some conveyor code
/:cl:

### CODE REVIEW

File: code/modules/recycling/conveyor2.dm [hunk 1]
```diff
@@ -193,16 +94,10 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
-
-/obj/machinery/conveyor/wirecutter_act(mob/living/user, obj/item/I)
-	if(machine_stat & BROKEN)
-		return TRUE
-	flipped = !flipped
+	conveyor_flags ^= CONVEYOR_INVERTED
 	update_move_direction()
-	to_chat(user, span_notice("You flip [src]'s belt [flipped ? "around" : "back to normal"]."))
+	update_icon()
+	to_chat(user, span_notice("You set [src]'s direction [conveyor_flags & CONVEYOR_INVERTED ? "backwards" : "back to default"]."))
```
Comment (TiviPlus) [inline code/modules/recycling/conveyor2.dm:100]: balloon alert?

File: code/modules/recycling/conveyor2.dm [hunk 2]
```diff
@@ -216,52 +111,149 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
-		C.set_operable(dir, id, 0)
+/obj/machinery/conveyor/process()
+	if(conveyor_flags & CONVEYOR_IS_CONVEYING)
+		return //you've made a lag monster
+	if(!is_operational())
+		return
+	if(!operating)
+		return
+	if(!isturf(loc))
+		return //how
```
Comment (TiviPlus) [inline code/modules/recycling/conveyor2.dm:126]: shouldnt some of these be returning PROCESS_KILL

File: code/modules/recycling/conveyor2.dm [hunk 3]
```diff
@@ -216,52 +111,149 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
+
+	//get the first 30 items in contents
+	var/turf/locturf = loc
```
Comment (TiviPlus) [inline code/modules/recycling/conveyor2.dm:129]: dunno why locturf is cast

File: code/modules/recycling/conveyor2.dm [hunk 4]
```diff
@@ -216,52 +111,149 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
...
-	var/obj/machinery/conveyor/C = locate() in get_step(src, stepdir)
-	if(C)
-		C.set_operable(stepdir, id, op)
+	if(operating)
+		start_processing()
+	else
+		stop_processing()
 
-/obj/machinery/conveyor/power_change()
+/obj/machinery/conveyor/proc/update()
```
Comment (TiviPlus) [inline code/modules/recycling/conveyor2.dm:214]: autodoc?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/recycling/conveyor2.dm (hunk 3 above)
Comment (Lumipharon) [reply code/modules/recycling/conveyor2.dm:129]: leftover thing

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**